File: evals/parity/reference/did.py
from diff_diff import CallawaySantAnna

res = CallawaySantAnna(control_group="never_treated", base_period="universal").fit(
    df, outcome="outcome", unit="unit", time="time",
    first_treat="first_treat", aggregate="all")
print(f"ATT:{res.overall_att}")

File: evals/parity/reference/did.R
suppressMessages(library(did))
att <- att_gt(yname = "outcome", tname = "time", idname = "unit", gname = "first_treat",
              data = df, control_group = "nevertreated", base_period = "universal",
              bstrap = FALSE)
ov <- aggte(att, type = "simple", na.rm = TRUE)
cat(sprintf("ATT:%f\n", ov$overall.att))
